core/src/main/java/org/infinispan/transaction/xa/TransactionFactory.java

```diff
@@ -251,11 +152,6 @@ protected long generateRandomId() {
          return rnd.nextLong();
```

Comment:
`Random` can be removed.
Also, `newRemoteTransaction()` can be removed from the `enum` since it is the same implementation in all values.

Reply:
`newRemoteTransaction()` implementations are different in `NODLD_RECOVERY_XA`